ctru-rs/examples/local-networking.rs

```diff
@@ -0,0 +1,301 @@
+//! Local networking example.
+//!
+//! This example showcases local networking using the UDS module.
+
+use ctru::prelude::*;
+use ctru::services::uds::*;
+
+fn handle_status_event(uds: &Uds, prev_node_mask: u16) -> ctru::Result<u16> {
+    println!("Connection status event signalled");
+    let status = uds.get_connection_status()?;
+    println!("Status: {status:#02X?}");
+    let left = prev_node_mask & (status.node_bitmask ^ prev_node_mask);
+    let joined = status.node_bitmask & (status.node_bitmask ^ prev_node_mask);
+    for i in 0..16 {
+        if left & (1 << i) != 0 {
+            println!("Node {} disconnected", i + 1);
+        }
+    }
+    for i in 0..16 {
+        if joined & (1 << i) != 0 {
+            println!(
+                "Node {} connected: {:?}",
+                i + 1,
+                uds.get_node_info(NodeID::Node(i + 1))
+            );
+        }
+    }
+    Ok(status.node_bitmask)
+}
+
+fn main() -> Result<(), Error> {
+    let apt = Apt::new().unwrap();
+    let mut hid = Hid::new().unwrap();
+    let gfx = Gfx::new().unwrap();
+    let console = Console::new(gfx.top_screen.borrow_mut());
+
+    println!("Local networking demo");
+
+    let mut uds = Uds::new(None).unwrap();
+
+    println!("UDS initialised");
+
+    enum State {
```

Comment:
This looks a lot like a manually implemented future state machine. could this have worked better with async?

Reply:
Very possibly! I'm scared of `async`.

Reply:
I've done some basic reading about futures, and I don't understand how this could be done with async. Could you explain how that would work? It would definitely be nice to clean up the code if it's possible.